parsers/SG.py

```diff
@@ -61,52 +54,61 @@
 }
 
 
-def get_solar(session=None):
+def get_solar(session, logger):
     """
     Fetches a graphic showing estimated solar production data.
     Uses OCR (tesseract) to extract MW value.
     Returns a float or None.
     """
 
-    s = session or requests.Session()
     url = 'https://www.ema.gov.sg/cmsmedia/irradiance/plot.png'
-    solar_image = Image.open(s.get(url, stream=True).raw)
+    solar_image = Image.open(session.get(url, stream=True).raw)
 
     gray = solar_image.convert('L')
-    threshold_filter = lambda x: 0 if x<77 else 255
+    threshold_filter = lambda x: 0 if x < 77 else 255
     black_white = gray.point(threshold_filter, '1')
 
     text = image_to_string(black_white, lang='eng')
 
-    pattern = r'Est. PV Output: (.*)MWac'
-    val = re.search(pattern, text, re.MULTILINE).group(1)
+    try:
```

Comment:
Much better handling, we shouldn't throw away data just because solar is not working.